src/main/cpp/src/get_json_object.cu

```diff
@@ -883,6 +902,49 @@ __device__ thrust::pair<bool, size_t> get_json_object_single(
   return {success, generator.get_output_len()};
 }
 
+/**
+ * @brief Kernel for running the JSONPath query.
+ *
+ * This kernel operates in a 2-pass way. On the first pass it computes the
+ * output sizes. On the second pass, it fills in the provided output buffers
+ * (chars and validity).
+ *
+ * @param col Device view of the incoming string
+ * @param commands JSONPath command buffer
+ * @param output_offsets Buffer used to store the string offsets for the results
+ *        of the query
+ * @param out_buf Buffer used to store the results of the query
+ * @param out_validity Output validity buffer
+ * @param out_valid_count Output count of # of valid bits
+ * @param options Options controlling behavior
+ */
+template <int block_size>
+__launch_bounds__(block_size) CUDF_KERNEL
+  void get_json_object_size_kernel(cudf::column_device_view col,
```

Comment:
Similar to the other overload, it is not necessary to have this kernel-I don't think we will gain performance by doing this. We can just use the existing kernel like before, but rewrite it a bit to minimize overhead when computing the output size. A few `if` instruction overhead in the kernel should not cause any performance difference since the kernel is limited by memory bandwidth, not compute capability

Reply:
done.

Reply:
To be clear the difference is about spilling and the number of registers. This is what I found when working on https://github.com/NVIDIA/spark-rapids-jni/pull/2015

By splitting them it reduced the number of registers for the one that does not write out data. You are correct that it has essentially no impact to the performance.